### PR TITLE
TES-415: Resource naming and more

### PR DESCRIPTION
**Pull request by mihailradkov**
## Changes

TES-415: Adhere to Azure naming conventions

Added abbreviation to each resource hinting at the resource type.

---

TES-413: Merged modules

- Moved the address module to the gateway
- Moved the identity module to the roles module

---

TES-388: Improved the GH workflow

- Added steps identifiers
- Replaced branches with exact SHAs
- Added global timeout of 5 minutes

---

Additionally

- Updated Bastion subnet cidr to allow more addresses
- Renamed the configuration module to configurations
- Renamed gateway module configurations for clarity
- Removed useless depends_on rules

### CODE REVIEW

File: modules/appconfig/main.tf [hunk 1]
```diff
@@ -8,7 +8,7 @@ resource "random_string" "app_config_name_prefix" {
 
 locals {
   # Creates an unique app configuration name to max of 50 characters
-  app_configuration_name = "${substr(var.resource_name_prefix, 0, 43)}-${random_string.app_config_name_prefix.result}"
+  app_configuration_name = "appcs-${substr(var.resource_name_prefix, 0, 37)}-${random_string.app_config_name_prefix.result}"
```
Comment (viktor-ribchev) [inline modules/appconfig/main.tf:11]: appcs - what the "s" for?

File: modules/bastion/main.tf [hunk 2]
```diff
@@ -77,15 +77,15 @@ resource "azurerm_subnet_network_security_group_association" "graphdb_bastion" {
...
+  name                = "pip-${var.resource_name_prefix}-bastion"
   location            = var.location
   resource_group_name = var.resource_group_name
   allocation_method   = "Static"
   sku                 = "Standard"
 }
 
 resource "azurerm_bastion_host" "graphdb" {
-  name                = "${var.resource_name_prefix}_bastion"
+  name                = "bas-${var.resource_name_prefix}"
```
Comment (viktor-ribchev) [inline modules/bastion/main.tf:88]: why not "bh" (bastion host) ? "bas" could be misleading

File: modules/appconfig/main.tf (hunk 1 above)
Comment (mihailradkov) [reply modules/appconfig/main.tf:11]: https://learn.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-abbreviations

File: modules/bastion/main.tf (hunk 2 above)
Comment (mihailradkov) [reply modules/bastion/main.tf:88]: https://learn.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-abbreviations

File: main.tf [hunk 3]
```diff
@@ -312,21 +281,25 @@ module "vmss" {
 
   resource_name_prefix = var.resource_name_prefix
   location             = var.location
+  resource_group_id    = azurerm_resource_group.graphdb.id
   resource_group_name  = azurerm_resource_group.graphdb.name
-  zones                = var.zones
+  zones                = [1, 3]
```
Comment (viktor-ribchev) [inline main.tf:286]: mistake?

Comment (mihailradkov) [reply main.tf]: already fixed..

**Review by yaskoo — APPROVED** [review]

**Review by viktor-ribchev — APPROVED** [review]

**Pull request merged**